doc/code/qml_fourier.rst

```diff
@@ -123,7 +123,227 @@ for multiple dimensions.
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While all the functions available for visualizing multiple sets of coefficients
+can be used for a single set, the primary tool for this purpose is the
+``fourier_bar_plot`` function.
+
+.. code::
+
+   from functools import partial
+   import matplotlib.pyplot as plt
+   import pennylane as qml
+   from pennylane.fourier import *
+
+   dev = qml.device('default.qubit', wires=2)
+
+   @qml.qnode(dev)
+   def simple_circuit(x):
+       qml.RY(x[0], wires=0)
+       qml.CNOT(wires=[1, 0])
+       qml.RX(x[0], wires=0)
+       return qml.expval(qml.PauliZ(0))
+
+   coeffs = fourier_coefficients(simple_circuit, 1, 2)
+
+   >>> coeffs
+   [0.5 +0.j 0.  +0.j 0.25+0.j 0.25+0.j 0.  +0.j]
+
+   
+   # Set up the axes
+   fig, ax = plt.subplots(2, 1, sharex=True, sharey=True)
+   fourier_bar_plot(coeffs, 1, ax)
+   plt.suptitle("Simple circuit bar plot")
+
+.. image:: ../_static/fourier_vis_bar_plot.png
+    :align: center
+    :width: 500px
+    :target: javascript:void(0);
+
+|
+	     
+In the bar plots, real coefficients are shown in the top panel, and complex in
+the bottom. The labels along the x-axis represent the coefficient frequencies
+(for large plots, it is sometimes convenient to remove these by passing
+``show_freqs=False`` to the plotting function.
+
+Below is a more complex example that demonstrates some of the additional
+customization options available:
+
+.. code::
+
+   weights = np.array([[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]])
+   
+   @qml.qnode(dev)
+   def circuit_with_weights(w, x):
+       qml.RX(x[0], wires=0)
+       qml.RY(x[1], wires=1)
+       qml.CNOT(wires=[1, 0])
+   
+       qml.Rot(*w[0], wires=0)
+       qml.Rot(*w[1], wires=1)
+       qml.CNOT(wires=[1, 0])
+       
+       qml.RX(x[0], wires=0)
+       qml.RY(x[1], wires=1)
+       qml.CNOT(wires=[1, 0])
+       
+       return qml.expval(qml.PauliZ(0))
+
+   coeffs = fourier_coefficients(partial(circuit_with_weights, weights), 2, 2)
+
+   # Number of inputs is now two; pass custom colours as well
+   fig, ax = plt.subplots(2, 1, sharex=True, sharey=True, figsize=(15, 4))
+   fourier_bar_plot(coeffs, 2, ax, colour_dict={"real" : "red", "imag" : "blue"});
+   plt.suptitle("Circuit with weights bar plot", fontsize=14)
+
+
+.. image:: ../_static/fourier_vis_bar_plot_2.png
+    :align: center 
+    :Width: 100%
+    :target: javascript:void(0);
+
+|
+	     
+Two convenience functions are also provided to visualize 1- and 2-dimensional
+functions given a set of Fourier coefficients:
+:func:`~.pennylane.fourier.fourier_reconstruct_function_1D_plot` and
+:func:`~.pennylane.fourier.fourier_reconstruct_function_2D_plot`. For example,
+``circuit_with_weights`` has two input parameters with respect to which we computed
+Fourier coefficients, so we can plot its output:
+
+.. code::
+
+   fourier_reconstruct_function_2D_plot(coeffs)
+   plt.title("Expectation value for circuit with weights", fontsize=14)
```

Comment:
I would add `x[0]` and `x[1]` here to the axes, so it is immediately clear what the numbers mean